raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py

```diff
@@ -28,30 +28,97 @@ def increasing_gas_price_strategy(web3: Web3, _transaction_params: Dict) -> GasP
-
-    second_proxy = second_client.new_contract_proxy(
-        abi=contract_proxy.contract.abi, contract_address=contract_proxy.contract_address
-    )
+    # Create a new instance of the JSONRPCClient, this will store the current available nonce
+    client_invalid_nonce = JSONRPCClient(web3=deploy_client.web3, privkey=deploy_client.privkey)
 
     check_block = deploy_client.get_checking_block()
     gas_estimate = contract_proxy.estimate_gas(check_block, "ret")
     assert gas_estimate, "Gas estimation should not fail here"
     startgas = safe_gas_limit(gas_estimate)
 
+    # At this point `client_invalid_nonce` has a nonce that is `1` too low,
+    # since a transaction was sent using `deploy_client` above and these two
+    # instances share the same underlying private key.
+    #
+    # Note that it is assumed this runs fast enough so that the first transaction is not
+    # mined before second is sent.
     contract_proxy.transact("ret", startgas)
-    with pytest.raises(TransactionAlreadyPending):
-        second_proxy.transact("ret", startgas)
+    with pytest.raises(EthereumNonceTooLow):
+        client_invalid_nonce.new_contract_proxy(
+            abi=contract_proxy.contract.abi, contract_address=contract_proxy.contract_address
+        ).transact("ret", startgas)
+
+
+def test_resending_mined_transaction_raises(deploy_client: JSONRPCClient) -> None:
+    """ If a mined transaction is re-sent the exception `EthereumNonceTooLow` is raised. """
+    # Use a _fixed_ gas price startegy so that both transactions are identical.
+    deploy_client.web3.eth.setGasPriceStrategy(make_fixed_gas_price_strategy(GasPrice(2000000000)))
+    contract_proxy, _ = deploy_rpc_test_contract(deploy_client, "RpcTest")
+
+    address = contract_proxy.contract_address
+    assert len(deploy_client.web3.eth.getCode(to_checksum_address(address))) > 0
+
+    # Create a new instance of the JSONRPCClient, this will store the current available nonce
+    client_invalid_nonce = JSONRPCClient(deploy_client.web3, deploy_client.privkey)
+
+    check_block = deploy_client.get_checking_block()
+    gas_estimate = contract_proxy.estimate_gas(check_block, "ret")
+    assert gas_estimate, "Gas estimation should not fail here"
+    startgas = safe_gas_limit(gas_estimate)
+
+    txhash = contract_proxy.transact("ret", startgas)
+    deploy_client.poll(txhash)
+
+    # At this point `client_invalid_nonce` has a nonce that is `1` too low,
+    # since a transaction was sent using `deploy_client` above and these two
+    # instances share the same underlying private key.
+    #
+    # Note that a the same function is called in this test.
+    with pytest.raises(EthereumNonceTooLow):
+        client_invalid_nonce.new_contract_proxy(
+            abi=contract_proxy.contract.abi, contract_address=contract_proxy.contract_address
+        ).transact("ret", startgas)
+
+
+def test_reusing_nonce_from_a_mined_transaction_raises(deploy_client: JSONRPCClient) -> None:
```

Comment:
The difference from this to `test_resending_mined_transaction_raises` is that this sends a different transaction.